wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -638,6 +652,31 @@ impl PhysicalDeviceFeatures {
             );
         }
 
+        if let Some(ref subgroup) = caps.subgroup {
+            if subgroup.supported_operations.contains(
+                vk::SubgroupFeatureFlags::BASIC
+                    | vk::SubgroupFeatureFlags::VOTE
+                    | vk::SubgroupFeatureFlags::ARITHMETIC
+                    | vk::SubgroupFeatureFlags::BALLOT
+                    | vk::SubgroupFeatureFlags::SHUFFLE
+                    | vk::SubgroupFeatureFlags::SHUFFLE_RELATIVE,
+            ) {
+                features.set(
+                    F::SUBGROUP,
+                    subgroup
+                        .supported_stages
+                        .contains(vk::ShaderStageFlags::COMPUTE | vk::ShaderStageFlags::FRAGMENT),
+                );
+                features.set(
+                    F::SUBGROUP_VERTEX,
+                    subgroup
+                        .supported_stages
+                        .contains(vk::ShaderStageFlags::VERTEX),
+                );
+                features.insert(F::SUBGROUP_BARRIER);
```

Comment:
We should only advertise those features if `VK_EXT_subgroup_size_control` is present or we are on vk 1.3.